extensions/funqy/funqy-knative-events/runtime/src/main/java/io/quarkus/funqy/runtime/bindings/knative/events/JsonCloudEventImpl.java

```diff
@@ -190,43 +190,45 @@ public T data() {
             }
         } else if (byte[].class.equals(dataType)) {
             try {
-                switch (specVersion().charAt(0)) {
-                    case '0':
-                        boolean isBase64 = false;
-                        if (event.has("datacontentencoding")) {
-                            String dce = event.get("datacontentencoding").asText();
-                            if ("base64".equals(dce)) {
-                                isBase64 = true;
-                            } else {
-                                throw new RuntimeException("Cannot deserialize data for data-content-encoding: '" + dce + "'.");
-                            }
+                if (specVersion() != null && specVersion().charAt(0) == '0') {
```

Comment:
Sorry, I didn't realize that 0.3 and 1.0 weren't backward compatible.  Just say so next time.  My bad.